iosdevapp/ViewModel.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  ModelView.swift
+//  iosdevapp
+//
+//  Created by Leonit Ejupi on 07.02.23.
+//
+
+import Foundation
+
+final class ViewModel: ObservableObject {
+    private let urlSession: URLSession
+    @Published var imageURL: URL?
+    @Published var isLoading = false
+    
+    init(urlSession: URLSession = URLSession.shared) {
+        self.urlSession = urlSession
```

Comment:
Let's respect the separation of concerns principle. I would suggest moving every detailed code about networking (`URLSession`, `URLRequest`, encoding and decoding) from the view model to `HTTPClient` and/or some `Service`.  Then you could use this service within your view model without knowing the details about the actual networking. It will make your code more structured, legible, and better testable.